qemu/tests/virtio_fs_multi_vms.py

```diff
@@ -65,148 +67,158 @@ def get_viofs_exe(session):
+                logging.info("%s: The guest file in shared dir is %s",
+                             vm, guest_file)
+                mapping[vm]['filesystems'].append({'fs_target': fs_target,
+                                                   'fs_dest': fs_dest,
+                                                   'guest_file': guest_file})
+
+                if cmd_dd:
+                    logging.info("%s: Creating file under %s inside guest.",
+                                 vm, fs_dest)
+                    session.cmd(cmd_dd % guest_file, io_timeout)
+
+                if shared_fs_source_dir:
+                    continue
 
                 if os_type == "linux":
                     error_context.context("%s: Umount the viriofs target %s." %
-                                          (vm, fs['fs_target']), logging.info)
-                    utils_disk.umount(fs['fs_target'], fs['fs_dest'],
-                                      'virtiofs', session=session)
-        if len(md5_set) != 1:
-            test.fail('The md5 values are different among VMs.')
+                                          (vm, fs_target), logging.info)
+                    utils_disk.umount(fs_target, fs_dest, 'virtiofs',
+                                      session=session)
+
+        if shared_fs_source_dir:
+            error_context.context("Compare the md5 among VMs.", logging.info)
+
+            md5_set = set()
+            for vm, info in mapping.items():
+                session = info['session']
+                for fs in info['filesystems']:
+                    shared_data = fs['guest_file']
+                    error_context.context("%s: Get the md5 of %s." %
+                                          (vm, shared_data), logging.info)
+                    if os_type == "linux":
+                        cmd_md5_vm = cmd_md5 % shared_data
+                    else:
+                        guest_file_win = shared_data.replace("/", "\\")
+                        cmd_md5_vm = cmd_md5 % (volume_letter, guest_file_win)
+
+                    md5_guest = session.cmd(cmd_md5_vm,
+                                            io_timeout).strip().split()[0]
+                    logging.info(md5_guest)
+                    md5_set.add(md5_guest)
+
+                    if os_type == "linux":
+                        error_context.context("%s: Umount the viriofs target %s." %
+                                              (vm, fs['fs_target']), logging.info)
+                        utils_disk.umount(fs['fs_target'], fs['fs_dest'],
+                                          'virtiofs', session=session)
+            if len(md5_set) != 1:
+                test.fail('The md5 values are different among VMs.')
+    finally:
+        for vm in vms:
+            if vm.is_alive():
+                vm.destroy()
+        for mem_path in [mem_path_vm1, mem_path_vm2, mem_path_vm3]:
+            process.run("rm -rf %s" % mem_path)
```

Comment:
I think we should not have do it here, instead of using `post_command` interface to let the framework to do that.